File: src/csharp/14_easy_最长公共前缀/S1.cs
/*
 * @lc app=leetcode.cn id=14 lang=csharp
 *
 * [14] 最长公共前缀
 */

namespace LeetCode.P14.S1;

// @lc code=start
using System.Text;

public class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        var resultSb = new StringBuilder();
        char now = ' ';
        for (var i = 0; i < strs.Min(s => s.Length); i++)
        {
            foreach (var str in strs)
            {
                if (str.Length < 0) goto end;
                if (now == ' ') now = str[i];
                else if (str[i] != now) goto end;
            }
            resultSb.Append(now);
            now = ' ';
        }

        end: return resultSb.ToString();
    }
}
// @lc code=end

File: src/python3/28_easy_找出字符串中第一个匹配项的下标/S1.py
#
# @lc app=leetcode.cn id=28 lang=python3
#
# [28] 找出字符串中第一个匹配项的下标
#


# @lc code=start
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        len_haystack = len(haystack)
        len_needle = len(needle)

        for i in range(len_haystack):
            if haystack[i] == needle[0] and len_haystack - i >= len_needle:
                if haystack[i : i + len_needle] == needle:
                    return i
        return -1


# @lc code=end
